src/js/_enqueues/lib/emoji-loader.js

```diff
@@ -2,65 +2,189 @@
+	/**
+	 * @typedef SessionSupportTests
+	 * @type {object}
+	 * @property {number} timestamp
+	 * @property {SupportTests} supportTests
+	 */
+
+	/**
+	 * Get support tests from session.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @private
+	 *
+	 * @returns {?SupportTests} Support tests, or null if not set or older than 1 day.
+	 */
+	function getSessionSupportTests() {
+		if (
+			typeof sessionStorage !== 'undefined' &&
+			sessionStorageKey in sessionStorage
+		) {
+			try {
+				/** @type {SessionSupportTests} */
+				var item = JSON.parse(
+					sessionStorage.getItem( sessionStorageKey )
+				);
+				if (
+					typeof item === 'object' &&
+					typeof item.timestamp === 'number' &&
+					new Date().valueOf() < item.timestamp + 86400 &&
+					typeof item.supportTests === 'object'
+				) {
+					return item.supportTests;
+				}
+			} catch ( e ) {}
+		}
+		return null;
+	}
+
+	/**
+	 * Persist the supports in session storage.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @private
+	 *
+	 * @param {SupportTests} supportTests Support tests.
+	 */
+	function setSessionSupportTests( supportTests ) {
+		if ( typeof sessionStorage !== 'undefined' ) {
+			try {
+				/** @type {SessionSupportTests} */
+				var item = {
+					supportTests: supportTests,
+					timestamp: new Date().valueOf()
+				};
+
+				sessionStorage.setItem(
+					sessionStorageKey,
+					JSON.stringify( item )
+				);
+			} catch ( e ) {}
+		}
+	}
 
 	/**
 	 * Checks if two sets of Emoji characters render the same visually.
 	 *
+	 * This function may be serialized to run in a Worker. Therefore, it cannot refer to variables from the containing
+	 * scope. Everything must be passed by parameters.
+	 *
 	 * @since 4.9.0
 	 *
 	 * @private
 	 *
+	 * @param {CanvasRenderingContext2D} context 2D Context.
 	 * @param {string} set1 Set of Emoji to test.
 	 * @param {string} set2 Set of Emoji to test.
 	 *
 	 * @return {boolean} True if the two sets render the same.
 	 */
-	function emojiSetsRenderIdentically( set1, set2 ) {
+	function emojiSetsRenderIdentically( context, set1, set2 ) {
 		// Cleanup from previous test.
-		context.clearRect( 0, 0, canvas.width, canvas.height );
+		context.clearRect( 0, 0, context.canvas.width, context.canvas.height );
 		context.fillText( set1, 0, 0 );
-		var rendered1 = canvas.toDataURL();
+		var rendered1 = new Uint32Array(
+			context.getImageData(
+				0,
+				0,
+				context.canvas.width,
+				context.canvas.height
+			).data
+		);
 
 		// Cleanup from previous test.
-		context.clearRect( 0, 0, canvas.width, canvas.height );
+		context.clearRect( 0, 0, context.canvas.width, context.canvas.height );
 		context.fillText( set2, 0, 0 );
-		var rendered2 = canvas.toDataURL();
+		var rendered2 = new Uint32Array(
+			context.getImageData(
+				0,
+				0,
+				context.canvas.width,
+				context.canvas.height
+			).data
+		);
 
-		return rendered1 === rendered2;
+		return rendered1.every( function ( rendered2Data, index ) {
+			return rendered2Data === rendered2[ index ];
+		} );
```

Comment:
This gets built right? Can we use arrow functions? 
```suggestion
		return rendered1.every( ( rendered2Data, index ) => rendered2Data === rendered2[ index ] );
```

Reply:
The file has to be ES3-compatible due to `grunt-jsvalidate`.

Reply:
See https://github.com/WordPress/wordpress-develop/pull/4562#discussion_r1232926948